### PR TITLE
{erlang,elixir}/generic-builder: /bin/sh fixes

### PR DESCRIPTION
**Pull request by pbogdan**
###### Motivation for this change

See commit messages.

- #36853
- #36823
- #37638

I would propose backporting to 18.03.

---
- [x] ran nox-review with `nixStable2` obtained from nixpkgs @ 
c5f141ff7a6403ca8abc3fcff90f13a8fe164fe4 (from before backports of `/bin/sh` fixes).
---

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/interpreters/erlang/generic-builder.nix [hunk 1]
```diff
@@ -66,6 +66,8 @@ in stdenv.mkDerivation ({
 
   postPatch = ''
     patchShebangs make
+    substituteInPlace erts/etc/unix/Install.src  \
+      --replace "#!/bin/sh" "${stdenv.shell}"
```
Comment (LnL7) [inline pkgs/development/interpreters/erlang/generic-builder.nix]: is the `#!` dropped here intentional?

**Comment by dezgeg** [comment]
If there still are hydra builders with the crap /bin/sh, the right thing is to complain more to get the hydra fixed.

Otherwise this is still broken on 17.09 for instance.

File: pkgs/development/interpreters/erlang/generic-builder.nix (hunk 1 above)
Comment (pbogdan) [reply pkgs/development/interpreters/erlang/generic-builder.nix]: Oh my.. Changed the match to be more precise and forgot to update the replacement.. Thank you for catching this!

**Comment by pbogdan** [comment]
As far as I can tell this is no longer needed.

**Comment by pbogdan** [comment]
As far as I can tell this is no longer needed.

**Pull request closed without merging**